File: C/readability.c
// Computes and print the approximate grade level needed to comprehend some text.
// by Wojciech Szubstarski 2021 for CS50.

#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float sentences = 0;
float words = 0;
float alphachars = 0;

int main(void)
{

    //Asks user for input.
    string text = get_string("Text: ");

    // Counts alpha characters, words and sentences in text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            alphachars++;
        }
        if (((i == 0) && (text[i] != ' ')) || ((i != (n - 1)) && (text[i] == ' ') && ((text[i + 1]) != ' ')))
        {
            words++;
        }
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            sentences++;
        }
    }


    // Calculates readability index with Coleman-Liau formula
    float L = 100 * alphachars / words;
    float S = 100 * sentences / words;
    int index = 0;
    index = round(0.0588 * L  - 0.296 * S - 15.8);

    // Prints approximate grade level needed to comprehend input text.
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Python/readability.py
# Computes and print the approximate grade level needed to comprehend some text.
# by Wojciech Szubstarski 2020 for CS50.
from cs50 import get_string

# Asks user for input.
text = get_string("Text: ")

sentences = 0
words = 0
alphachars = 0

# Counts aplha characters, words and sentences in text
for i in range(len(text)):
    if text[i].isalpha():
        alphachars += 1
    if (i == 0 and text[i] != " ") or (i != len(text) - 1 and text[i] == " " and text[i + 1] != " "):
        words += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

# Calculates readability index with Coleman-Liau formula
index = round((0.0588 * (alphachars / words * 100)) - (0.296 * (sentences / words * 100)) - 15.8)

# Prints approximate grade level needed to comprehend input text.
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", + index)